default/python/universe_generation/empires.py

```diff
@@ -80,6 +81,26 @@ def count_planets_in_systems(systems, planet_types_filter=HS_ACCEPTABLE_PLANET_T
     return num_planets
 
 
+def list_planet_types_in_systems(systems):
+    """
+    Get a list of planet types in systems
+    """
+    planet_types_in_systems = []
+    for system in systems:
+        for planet in fo.sys_get_planets(system):
+            planet_types_in_systems.append(fo.planet_get_type(planet))
+    return planet_types_in_systems
```

Comment:
You could also simplify this one:

Keep in mind that this is changing the return type to set as well.  I don't think that list has since for this data, but if you want to keep it for compatibility just replace `{}` with `[]`

Also, I recommend to keep the function, even if it's a one-line.  Expression is a bit complex, so it's good to have a name for it.
```suggestion
    return {fo.planet_get_type(planet) for system in systems for planet in fo.sys_get_planets(system)}
```

Maybe you will be able to simplify count_planet_types_in_systems after that.